src/util/http_utils.js

```diff
@@ -946,6 +946,40 @@ function set_response_headers_from_request(req, res) {
     if (req.query['response-expires']) res.setHeader('Expires', req.query['response-expires']);
 }
 
+/**
+ * autheticate jwt token for prometheus metrics and version request
+ * @param {nb.S3Request} req
+ * @param {nb.S3Response} res
+ * @param {string[]} roles
+ */
+function authorize_bearer(req, res, roles) {
+    let token = req.headers.authorization;
+    try {
+        if (!token) {
+            dbg.error('Missing authorization header : ', token);
+            throw new S3Error(S3Error.AccessDenied);
+        }
+        if (token.includes('Bearer ')) {
+            token = token.split(' ')[1];
+        }
+        const decoded = jwt_utils.authorize_jwt_token(token);
+        // Role 'metrics-auth' is used in operator RPC call, 
+        // If needs to change update operator RPC call first
+        if (!roles.includes(decoded.role)) {
+            dbg.error('Bearer token authorization failed : ', token);
+            throw new S3Error(S3Error.AccessDenied);
+        }
+    } catch (err) {
+        dbg.error('JWT verification failed for token : ', token, err);
+        res.writeHead(403, { 'Content-Type': 'application/json' });
+        const reply = JSON.stringify({
+            error: 'AccessDenied',
+            message: err.message,
+        }, null, 2);
+        res.end(reply);
```

Comment:
you need to return something like false from here or throw, otherwise the caller will continue to process the request too